verifiers/internal/gha/verifier.go

```diff
@@ -255,10 +255,19 @@ func (v *GHAVerifier) VerifyImage(ctx context.Context,
 		return nil, nil, err
 	}
 
-	// Parse any provenance target repository set using environment variable COSIGN_REPOSITORY
-	provenanceTargetRepository, err := ociremote.GetEnvTargetRepository()
-	if err != nil {
-		return nil, nil, err
+	var provenanceTargetRepository name.Repository
+	// Consume input for --provenance-repository when set
+	if *provenanceOpts.ExpectedProvenanceRepository != "" {
+		provenanceTargetRepository, err = name.NewRepository(*provenanceOpts.ExpectedProvenanceRepository)
+		if err != nil {
+			return nil, nil, err
+		}
+	} else {
+		// If user input --provenance-repository is empty, look for COSIGN_REPOSITORY environment
```

Comment:
I was about to suggest this as well since we'll probably want to move to using `sigstore-go` and not rely on cosign packages. The `--provenance-repository` option is a better way to implement what we want.